### PR TITLE
Infinex core staging deployment

### PR DESCRIPTION
**Pull request by ghost**
The arbitrum and base deployments don't have the accept ownership steps because staging transactions wasn't working (and still isn't). The steps have been added for the other chains but awaiting on cannon fix it so they can be staged.

### CODE REVIEW

**Comment by zeus-infinex** [comment]
https://sys.infinex.xyz/reports/evm-contracts/deployments-test/8c5d526/report/testnets/
# Testnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/16/2024, 3:34:20 AM | 5.191 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/8c5d526/report/staging/
# Staging Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/16/2024, 3:34:20 AM | 5.291 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/8c5d526/report/mainnets/
# Mainnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/16/2024, 3:34:20 AM | 20.601 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|0|0|0|8|
|Tests|120|5|5|0|130|

**Comment by zeus-infinex** [comment]
https://sys.infinex.xyz/reports/evm-contracts/deployments-test/8959444/report/testnets/
# Testnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/16/2024, 3:39:01 AM | 4.858 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/8959444/report/staging/
# Staging Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/16/2024, 3:39:01 AM | 5.017 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/8959444/report/mainnets/
# Mainnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/16/2024, 3:39:01 AM | 4.841 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

**Comment by zeus-infinex** [comment]
https://sys.infinex.xyz/reports/evm-contracts/deployments-test/d8d4cb0/report/testnets/
# Testnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/19/2024, 2:43:11 AM | 5.148 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/d8d4cb0/report/staging/
# Staging Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/19/2024, 2:43:10 AM | 5.392 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/d8d4cb0/report/mainnets/
# Mainnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/19/2024, 2:43:11 AM | 4.404 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

**Comment by zeus-infinex** [comment]
https://sys.infinex.xyz/reports/evm-contracts/deployments-test/2268694/report/testnets/
# Testnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/19/2024, 4:43:48 AM | 5.3 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/2268694/report/staging/
# Staging Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/19/2024, 4:43:47 AM | 5.225 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/2268694/report/mainnets/
# Mainnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/19/2024, 4:43:48 AM | 4.647 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

**Comment by zeus-infinex** [comment]
https://sys.infinex.xyz/reports/evm-contracts/deployments-test/4fe9571/report/testnets/
# Testnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/19/2024, 4:58:55 AM | 4.697 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/4fe9571/report/staging/
# Staging Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/19/2024, 4:58:55 AM | 5.099 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/4fe9571/report/mainnets/
# Mainnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/19/2024, 4:58:55 AM | 5.305 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

**Comment by zeus-infinex** [comment]
https://sys.infinex.xyz/reports/evm-contracts/deployments-test/e66a938/report/testnets/
# Testnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/20/2024, 3:50:28 AM | 4.599 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/e66a938/report/staging/
# Staging Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/20/2024, 3:50:28 AM | 4.485 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/e66a938/report/mainnets/
# Mainnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/20/2024, 3:50:28 AM | 4.464 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

**Review by axefinaxe — DISMISSED** [review]

**Comment by axefinaxe** [comment]
nits about the naming @rayInfinex , as discussed, happy to re-approve

**Comment by zeus-infinex** [comment]
https://sys.infinex.xyz/reports/evm-contracts/deployments-test/2b311d1/report/testnets/
# Testnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/20/2024, 4:44:38 AM | 4.919 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/2b311d1/report/staging/
# Staging Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/20/2024, 4:44:38 AM | 4.73 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/2b311d1/report/mainnets/
# Mainnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/20/2024, 4:44:38 AM | 4.415 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

**Comment by zeus-infinex** [comment]
https://sys.infinex.xyz/reports/evm-contracts/deployments-test/e9754e4/report/testnets/
# Testnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/21/2024, 11:06:02 AM | 4.655 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/e9754e4/report/staging/
# Staging Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/21/2024, 11:06:02 AM | 5.173 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/e9754e4/report/mainnets/
# Mainnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/21/2024, 11:06:02 AM | 4.444 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

**Comment by zeus-infinex** [comment]
https://sys.infinex.xyz/reports/evm-contracts/deployments-test/b7ff2cc/report/testnets/
# Testnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/22/2024, 1:49:28 AM | 5.267 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/b7ff2cc/report/staging/
# Staging Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/22/2024, 1:49:28 AM | 4.712 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/b7ff2cc/report/mainnets/
# Mainnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/22/2024, 1:49:28 AM | 4.346 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

**Comment by rayInfinex** [comment]
The arbitrum and base deployments don't have the accept ownership steps because staging transactions wasn't working (and still isn't). The steps have been added for the other chains but awaiting on cannon fix it so they can be staged.

File: scripts/migration/testnets/set-trusted-recovery-keeper.toml
Comment (rayInfinex) [inline scripts/migration/testnets/set-trusted-recovery-keeper.toml:1]: this was just a script used to update the testnet deployment.

File: infinex-multichain/staging/infinex-curve-app-test.ethereum.toml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+options.CURVE_STABLESWAP_SALT = "CURVE_STABLESWAP_1_1_0staging"
+options.CURVE_STABLESWAP_OWNER = "<%= settings.DEPLOYER %>"
+options.CURVE_STABLESWAP_USDC = "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48"
+options.CURVE_STABLESWAP_FACTORY = "0x6A8cbed756804B16E05E741eDaBd5cB544AE21bf"
+source = "infinex:1.1.0@curve-stableswap"
+target = "infinex:1.1.0@curve-stableswap"
+
+# ---------- Add To Registry ---------- #
+
+[pull.InfinexCoreTest]
```
Comment (rayInfinex) [inline infinex-multichain/staging/infinex-curve-app-test.ethereum.toml:23]: this was done to show what it looks like, but we won't be doing this going forwards.

File: infinex-multichain/staging/infinex-curve-app-test.ethereum.toml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+source = "infinex-core-test:latest@core"
```
Comment (axefinaxe) [inline infinex-multichain/staging/infinex-curve-app-test.ethereum.toml:24]: we should not be using "latest" cc @lemoninfinex

**Review by axefinaxe — CHANGES_REQUESTED** [review]

File: scripts/migration/staging/1-set-app-registry.toml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# pnpm cannon build scripts/migration/staging/1-set-app-registry.toml --dry-run --chain-id
+
+name = "infinex-test-set-app-registry"
+version = "3.0.0"
+
+[pull.InfinexCoreTest]
+source = "infinex-core-test:latest@core"
+
+[pull.Curve]
+source = "infinex-curve-app-test:latest@curve-stableswap"
```
Comment (axefinaxe) [inline scripts/migration/staging/1-set-app-registry.toml:11]: same

File: scripts/migration/staging/1-set-app-registry.toml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Add the apps to the app registry
+# pnpm cannon build scripts/migration/staging/1-set-app-registry.toml --dry-run --chain-id
+
+name = "infinex-test-set-app-registry"
+version = "3.0.0"
+
+[pull.InfinexCoreTest]
+source = "infinex-core-test:latest@core"
```
Comment (axefinaxe) [inline scripts/migration/staging/1-set-app-registry.toml:8]: same

**Comment by zeus-infinex** [comment]
https://sys.infinex.xyz/reports/evm-contracts/deployments-test/9719b0e/report/testnets/
# Testnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/22/2024, 4:31:40 AM | 5.29 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/9719b0e/report/staging/
# Staging Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/22/2024, 4:31:39 AM | 5.035 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/9719b0e/report/mainnets/
# Mainnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/22/2024, 4:31:40 AM | 4.314 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

**Comment by zeus-infinex** [comment]
https://sys.infinex.xyz/reports/evm-contracts/deployments-test/7417cfc/report/testnets/
# Testnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/22/2024, 7:25:38 AM | 5.209 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/7417cfc/report/staging/
# Staging Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/22/2024, 7:25:37 AM | 4.57 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/7417cfc/report/mainnets/
# Mainnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/22/2024, 7:25:38 AM | 5.039 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

**Comment by zeus-infinex** [comment]
https://sys.infinex.xyz/reports/evm-contracts/deployments-test/2f3a00c/report/testnets/
# Testnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/22/2024, 8:05:26 AM | 5.418 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/2f3a00c/report/staging/
# Staging Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/22/2024, 8:05:26 AM | 5.477 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

https://sys.infinex.xyz/reports/evm-contracts/deployments-test/2f3a00c/report/mainnets/
# Mainnets Report

| 🕙 Start time | ⌛ Duration |
| --- | ---: |
| 8/22/2024, 8:05:26 AM | 4.659 s |

| | ✅ Passed | ❌ Failed | ⏩ Skipped | 🚧 Todo | ⚪ Total |
| --- | ---: | ---: | ---: | ---: | ---: |
|Test Suites|2|1|0|0|3|
|Tests|10|0|0|0|10|

File: infinex-multichain/staging/infinex-curve-app-test.arbitrum.toml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+[var.Deploy]
+DEPLOYER = "0x6298551D56F825B3b6c3350D32129Ffd3cFb198C"
+
+[clone.CurveStableSwap]
+options.CURVE_STABLESWAP_SALT = "CURVE_STABLESWAP_1_1_0staging"
+options.CURVE_STABLESWAP_OWNER = "<%= settings.DEPLOYER %>"
+options.CURVE_STABLESWAP_USDC = "0xaf88d065e77c8cC2239327C5EDb3A432268e5831"
+options.CURVE_STABLESWAP_FACTORY = "0x9AF14D26075f142eb3F292D5065EB3faa646167b"
+source = "infinex:1.1.0@curve-stableswap"
```
Comment (axefinaxe) [inline infinex-multichain/staging/infinex-curve-app-test.arbitrum.toml:18]: Incorrect location

**Review by axefinaxe — APPROVED** [review]

**Pull request merged**